src/simpledb/query/Predicate.java

```diff
@@ -132,17 +132,17 @@ public Predicate joinSubPred(Schema sch1, Schema sch2) {
 
    /**
     * Determine if there is a term of the form "F=c" where F is the specified field
-    * and c is some constant. If so, the method returns that constant. If not, the
-    * method returns null.
+    * and c is some constant. If so, the method returns that Term (ie. Constant and
+    * Operator). If not, the method returns null.
     * 
     * @param fldname the name of the field
-    * @return either the constant or null
+    * @return either the term or null
     */
-   public Constant equatesWithConstant(String fldname) {
+   public Term equatesWithConstant(String fldname) {

```

Comment:
dont get this change

Reply:
This is to give access to the operator of the Term (rather than just the constant itself) so that tableplanner can choose to use idx iff the operator is `=`